core/domain/exp_services_test.py

```diff
@@ -924,6 +924,17 @@ def setUp(self):
             'param_changes': []
         }
 
+    def test_add_state_name(self):
+        """Test adding of state"""
+        exploration = exp_services.get_exploration_by_id(self.EXP_ID)
+        exp_services.update_exploration(self.owner_id, self.EXP_ID, [{
+            'cmd': 'add_state',
+            'state_name': 'new state',
+        }], 'Add state name')
+
+        exploration = exp_services.get_exploration_by_id(self.EXP_ID)
+        self.assertIn('new state', exploration.states)
```

Comment:
Tests are a lot stronger if they follow the following format:

1. Assert the current state
2. Change something
3. Assert the new state

This test is lacking step (1), so there's no guarantee from reading this code that ``update_exploration`` actually needs to do anything. The exploration could already have the state ``new_state`` and pass this test (if ``update_exploration`` didn't do anything). Can you please make sure all tests in this file calling ``update_exploration`` follow this format?